server/middlewares/validators/videos/video-passwords.ts

```diff
@@ -0,0 +1,89 @@
+import express from 'express'
+import { MUserAccountUrl } from '@server/types/models'
+import { HttpStatusCode, UserRight } from '@shared/models'
+import { MVideoFullLight } from '../../../types/models/video'
+import {
+  areValidationErrors,
+  doesVideoExist,
+  isVideoPasswordProtected,
+  isValidVideoIdParam,
+  doesVideoPasswordExist,
+  isVideoPasswordDeletable
+} from '../shared'
+import { body, param } from 'express-validator'
+import { isIdValid } from '@server/helpers/custom-validators/misc'
+import { isValidPasswordProtectedPrivacy } from '@server/helpers/custom-validators/videos'
+
+const listVideoPasswordValidator = [
+  isValidVideoIdParam('videoId'),
+
+  async (req: express.Request, res: express.Response, next: express.NextFunction) => {
+    if (areValidationErrors(req, res)) return
+
+    if (!await doesVideoExist(req.params.videoId, res, 'only-video')) return
+    if (!isVideoPasswordProtected(res.locals.onlyVideo, res)) return
+
+    return next()
+  }
+]
+
+const updateVideoPasswordListValidator = [
+  body('passwords'),
+
+  async (req: express.Request, res: express.Response, next: express.NextFunction) => {
+    if (areValidationErrors(req, res)) return
+
+    if (!await doesVideoExist(req.params.videoId, res, 'only-video')) return
+    if (!isValidPasswordProtectedPrivacy(req, res)) return
+
+    return next()
+  }
+]
+
+const removeVideoPasswordValidator = [
+  isValidVideoIdParam('videoId'),
+
+  param('passwordId')
+    .custom(isIdValid),
+
+  async (req: express.Request, res: express.Response, next: express.NextFunction) => {
+    if (areValidationErrors(req, res)) return
+
+    if (!await doesVideoExist(req.params.videoId, res, 'only-video')) return
+    if (!isVideoPasswordProtected(res.locals.onlyVideo, res)) return
+    if (!await doesVideoPasswordExist(req.params.passwordId, res.locals.onlyVideo, res)) return
+    if (!await isVideoPasswordDeletable(res.locals.videoPassword, res.locals.onlyVideo, res)) return
+
+    // Check if the user who did the request is able to delete the video passwords
+    if (!checkUserCanDeleteVideoPasswords(res.locals.oauth.token.User, res.locals.videoAll, res)) return
```

Comment:
Maybe you can merge `isVideoPasswordDeletable` in `checkUserCanDeleteVideoPasswords`?